src/json-patch-duplex.ts

```diff
@@ -145,66 +154,43 @@ module jsonpatch {
   /* The operations applicable to object root. Many are the same as for the object */
   var rootOps = {
     add: function (obj) {
-      rootOps.remove.call(this, obj);
+      var removed = rootOps.remove.call(this, obj);
       for (var key in this.value) {
```

Comment:
I was speaking about rewriting object property by property, not the `rootOps` in general